### PR TITLE
[NCL-8549] Fix GitLab scm url redirection on Build Detail page

### PR DESCRIPTION
**Pull request by DnsZhou**

### CODE REVIEW

File: src/utils/urlParseHelper.ts [hunk 1]
```diff
@@ -15,15 +15,20 @@ export interface IParsedUrl {
...
     : url.split(base + (['https', 'http'].includes(protocol) ? '/gerrit/' : '/'))[1];
-  const webUrl = url.includes('git@')
-    ? 'https://' + base + '/' + project
-    : 'https://' + base + '/gerrit/gitweb?p=' + project + ';a=summary';
-  const name = url.includes('git@') ? 'GitLab' : 'Gerrit';
+  const webUrl =
+    isGitProtocol || url.includes(protocol + '://' + gitlabBase)
+      ? 'https://' + base + '/' + project
+      : 'https://' + base + '/gerrit/gitweb?p=' + project + ';a=summary';
+  const name = url.includes('git@') || url.includes(protocol + '://' + gitlabBase) ? 'GitLab' : 'Gerrit';
```
Comment (patrikk0123) [inline src/utils/urlParseHelper.ts:31]: `url.includes('git@') || url.includes(protocol + '://' + gitlabBase)`
is computed twice, move it into variable (isGitlabUrl)
also `url.includes(gitlabBase)` might be enough

File: src/utils/urlParseHelper.ts [hunk 2]
```diff
@@ -15,15 +15,20 @@ export interface IParsedUrl {
...
-  const project = url.includes('git@')
+  const gitlabBase = 'gitlab.cee.redhat.com';
+  const isGitProtocol = url.includes('git@');
+  const protocol = isGitProtocol ? 'git@' : url.split('://')[0];
+  const base = isGitProtocol ? url.split('@').at(1)?.split(':').at(0) : url.split('://')[1].split('/')[0];
+  const project = isGitProtocol
     ? url.split(':').at(1)
+    : url.includes(gitlabBase)
+    ? url.split(gitlabBase + '/').at(1)
     : url.split(base + (['https', 'http'].includes(protocol) ? '/gerrit/' : '/'))[1];
```
Comment (patrikk0123) [inline src/utils/urlParseHelper.ts:26]:
```ts
const project = isGitProtocol
    ? url.split(':').at(1)
    : url.split(base + url.includes(gitlabBase)  ? '/' : '/gerrit/')).at(1);
```

File: src/utils/urlParseHelper.ts [hunk 3]
```diff
@@ -15,15 +15,20 @@ export interface IParsedUrl {
  * @returns Object containing scmRepository URL, parsed URL and display name representing URL
  *  */
 export const parseInternalScmRepositoryUrl = ({ url }: IScmRepositoryUrl): IParsedUrl => {
-  const protocol = url.includes('git@') ? 'git@' : url.split('://')[0];
-  const base = url.includes('git@') ? url.split('@').at(1)?.split(':').at(0) : url.split('://')[1].split('/')[0];
-  const project = url.includes('git@')
+  const gitlabBase = 'gitlab.cee.redhat.com';
+  const isGitProtocol = url.includes('git@');
+  const protocol = isGitProtocol ? 'git@' : url.split('://')[0];
+  const base = isGitProtocol ? url.split('@').at(1)?.split(':').at(0) : url.split('://')[1].split('/')[0];
```
Comment (patrikk0123) [inline src/utils/urlParseHelper.ts:21]: while you are at it, please convert all [] to .at()

File: src/utils/urlParseHelper.ts (hunk 2 above)
Comment (DnsZhou) [reply src/utils/urlParseHelper.ts:26]: This refactored code dose not have the same logic as the original one and will cause failure of parsing the URL start with `https://gitlab.***.com/` without `/gerrit/`. I will keep the original logic but with all [] change to at().

File: src/utils/urlParseHelper.ts [hunk 4]
```diff
@@ -15,15 +15,20 @@ export interface IParsedUrl {
  * @returns Object containing scmRepository URL, parsed URL and display name representing URL
  *  */
 export const parseInternalScmRepositoryUrl = ({ url }: IScmRepositoryUrl): IParsedUrl => {
-  const protocol = url.includes('git@') ? 'git@' : url.split('://')[0];
-  const base = url.includes('git@') ? url.split('@').at(1)?.split(':').at(0) : url.split('://')[1].split('/')[0];
-  const project = url.includes('git@')
+  const gitlabBase = 'gitlab.cee.redhat.com';
+  const isGitProtocol = url.includes('git@');
+  const protocol = isGitProtocol ? 'git@' : url.split('://').at(0)!;
```
Comment (patrikk0123) [inline src/utils/urlParseHelper.ts:20]: don't use `!`
rather provide fallback `?? ""`

**Review by patrikk0123 — COMMENTED** [review]
Maybe the Gerrit support can be removed, but I have no answer regarding that yet.

**Comment by DnsZhou** [comment]
> Maybe the Gerrit support can be removed, but I have no answer regarding that yet.

According to the planning meeting today, Kuba decided that we should not yet remove all support to Gerrit as we are not very sure if we can migrate to GitLab before the new UI is released. cc @patrikk0123

**Review by patrikk0123 — APPROVED** [review]

**Pull request merged**